Source/Core/SerializedMapProjection.js

```diff
@@ -0,0 +1,108 @@
+define([
+        '../ThirdParty/when',
+        './Check',
+        './CustomProjection',
+        './defined',
+        './DeveloperError',
+        './Ellipsoid',
+        './freezeObject',
+        './GeographicProjection',
+        './Proj4Projection',
+        './Rectangle',
+        './WebMercatorProjection'
+    ], function(
+        when,
+        Check,
+        CustomProjection,
+        defined,
+        DeveloperError,
+        Ellipsoid,
+        freezeObject,
+        GeographicProjection,
+        Proj4Projection,
+        Rectangle,
+        WebMercatorProjection) {
+    'use strict';
+
+    var ProjectionType = freezeObject({
+        GEOGRAPHIC : 0,
+        WEBMERCATOR : 1,
+        PROJ4JS : 2,
+        CUSTOM : 3
+    });
+
+    /**
+     * Serializes and Deserializes MapProjections.
+     *
+     * @param {MapProjection} mapProjection Any MapProjection supported by Cesium.
+     * @alias SerializedMapProjection
+     * @constructor
+     * @private
+     */
+    function SerializedMapProjection(mapProjection) {
```

Comment:
Also, why is this a separate class and not just part of the projection interface like it is for the geometry and basic types?  Also, since we now have 4 projections, shouldn't we be defining a `Projection` base interface so that the API is formally documented?

Reply:
Nevermind, I see we already have `MapProjection` as the base interface, why isn't pack part of that?

Reply:
> why isn't pack part of that?

That makes sense, I'll give it a try. For now I'm probably still going to just generate JSON instead of actual packed values though, since we're probably going to have to attach an enum to the message to indicate how to unpack on the worker-side.